cumulus_etl/common.py

```diff
@@ -37,10 +39,14 @@ def _atomic_open(path: str, mode: str) -> TextIO:
     """A version of open() that handles atomic file access across many filesystems (like S3)"""
     root = store.Root(path)
 
-    # fsspec is atomic per-transaction -- if an error occurs inside the transaction, partial writes will be discarded
-    with root.fs.transaction:
-        with root.fs.open(path, mode=mode, encoding="utf8") as file:
-            yield file
+    with contextlib.ExitStack() as stack:
+        if "w" in mode:
+            # fsspec is atomic per-transaction.
+            # If an error occurs inside the transaction, partial writes will be discarded.
+            # But we only want a transaction if we're writing - read transactions may error out
+            stack.enter_context(root.fs.transaction)
```

Comment:
I don't know why we weren't hitting this before - I was getting test errors on I think fake memory/buffer input files that didn't expect certain fsspec semantics - but it makes sense to skip the transaction on read anyway.